internal/tablemap.go

```diff
@@ -127,3 +127,43 @@ func GetSpannerCols(conv *Conv, srcTable string, srcCols []string) ([]string, er
 	}
 	return spCols, nil
 }
+
+// GetSpannerKeyName maps source foreign key name to
+// legal Spanner foreign key name
+// If the srcKeyName is empty string we can just return
+// empty string without error.
+// If the srcKeyName is not empty we need to make sure
+// of the following things:
+// a) the new foreign key name is legal
+// b) the new foreign key name doesn't clash with other Spanner
+//    foreignkey names
+// c) we consistently return the same key name.
+func GetSpannerKeyName(conv *Conv, srcKeyName string, schemaForeignKeys map[string]bool) (string, error) {
+	if srcKeyName == "" {
+		return "", nil
+	}
+	spKeyName, _ := FixName(srcKeyName)
+
+	// spKeyName has been used before i.e. FixName caused a collision.
+	// Add unique postfix: use number of foreign keys so far.
+	// However, there is a chance this has already been used,
+	// so need to iterate.
+	if _, found := schemaForeignKeys[spKeyName]; found {
+		id := len(schemaForeignKeys)
+		for {
+			c := spKeyName + "_" + strconv.Itoa(id)
+
+			// check wether the spKeyName is used before
+			if _, found := schemaForeignKeys[c]; !found {
+				spKeyName = c
+				break
+			}
+			id++
+		}
+	}
+
+	if spKeyName != "" {
```

Comment:
Why is this needed?
We've already handled the empty string case at the start of the function?

If it can be deleted, then would be more compact to simply return from the body of the if loop i.e. instead of the break at line 159, just have:

             return c, nil

and we can remove line 158 as well.

Reply:
we need to set the schemaForeignKeys[spKeyName]=true to note that this spanner key name is used.
but we don't need the condition, so removed it

Reply:
Ah, right -- thanks!